mediorum/server/serve_blob.go

```diff
@@ -167,6 +167,29 @@ func (ss *MediorumServer) getBlob(c echo.Context) error {
 	return c.String(404, "blob not found")
 }
 
+func (ss *MediorumServer) headBlob(c echo.Context) error {
+	ctx := c.Request().Context()
+	key := c.Param("cid")
+	logger := ss.logger.With("cid", key)
+
+	if ss.isCidBlacklisted(ctx, key) {
+		logger.Info("cid is blacklisted")
+		return c.String(403, "cid is blacklisted by this node")
+	}
+
+	// Pretend legacy exists for HEAD, and let it fail in the GET (getBlob) if it doesn't. Otherwise we have to duplicate a db query
+	if isLegacyCID(key) {
+		return c.NoContent(200)
+	}
+
+	// If the blob doesn't exist, return 404. This is similar to what the GET (getBlob) does
+	if attrs, err := ss.bucket.Attributes(ctx, key); err == nil && attrs != nil {
+		return c.NoContent(200)
+	}
+
+	return c.String(404, "blob not found")
```

Comment:
does the status code resp matter for these head requests? bc in the real `GET /tracks/cidstream` dont we 302 redirect before 404ing?

Reply:
for clients, no I don't think it matters. all the 404 HEADs are coming from one place according to CloudFlare (probably UptimeRobot like @alecsavvy noticed), so it's good to at least fix it for the sake of this consumer and our logs/graphs

<img width="924" alt="Screenshot 2023-06-22 at 10 31 02 PM" src="https://github.com/AudiusProject/audius-protocol/assets/4657956/59540be8-fe93-4d51-bad4-880d530d4418">


I couldn't decide if it was bad to do the whole db check in the HEAD since it would just be duplicated in the GET, but there's db cache and I realized actually if the HEAD returns 302 then the GET will just go straight to the correct node from that. so I changed that now -- good call